pending-documentation.md

```diff
@@ -648,6 +648,72 @@ ambiguity during parsing, so care should be taken to ensure that named
 nonterminals are sufficiently unique from one another to prevent such
 ambiguities. Of course, the compiler will generate a warning in this case.
 
+### `concrete` attribute, `#isConcrete` and `#isVariable` function
+
+**NOTE**: `#isConcrete` and `#isVariable` are _not_ supported by the Haskell
+backend.
+
+Sometimes you only want a given function to simplify if all (or some) of the
+arguments are concrete (non-symbolic). To do so, you can use either the
+`concrete` attribute (if you want it to only apply when all arguments are
+concrete), or the `#isConcrete(_)` side-condition (when you only want it to
+apply if some arguments are concrete). Conversly, the function `#isVariable(_)`
+will only return true when the argument is a variable.
+
+For example, the following will only re-associate terms when all arguments
+are concrete:
+
+```k
+rule X +Int (Y +Int Z) => (X +Int Y) +Int Z [concrete]
+```
+
+And the following rules will only re-associate terms when it will end up
+grouping concrete sub-terms:
+
+```k
+rule X +Int (Y +Int Z) => (X +Int Y) +Int Z
+  requires #isConcrete(X)
+   andBool #isConcrete(Y)
+   andBool #isVariable(Z)
+
+rule X +Int (Y +Int Z) => (X +Int Z) +Int Y
+  requires #isConcrete(X)
+   andBool #isConcrete(Z)
+   andBool #isVariable(Y)
+```
+
+### `simplification` attribute
+
+The simplification attribute identifies axioms that are useful for simplifying
+configurations, without being part of the main semantics. When a function rule
+is tagged as `simplification`, the Haskell backend will only apply that rule if
+the side condition has no remainder given the current top-level predicate (that
```

Comment:
And if the rule _matches_ the configuration.

Reply:
I was under the impression that the haskell backend _always_ uses matching for functions? So I didn't want to put it here, because it might confuse people who take this to mean that usually it's not using matching.

Reply:
Simplification rules are not function rules: they are not restricted to function-like equalities (in fact, they may not have function symbols at their head at all) and they are not included when closing over the definition rules for `owise`. Unlike definition rules, which are applied as a group, simplification rules are attempted one-by-one.

Furthermore, users should not assume that function rules are applied by matching. The operational semantics of function evaluation is undefined, the backend may do whatever it likes. It happens to use matching at this time, but we reserve the right to change that without notice. (For example, we may decide to automatically simplify `\ceil(function)` expressions, which would probably require using unification.) On the other hand, using matching is part of the contract of the `simplification` attribute.

Reply:
I believe I've captured this.